pennylane/devices/default_qubit_torch.py

```diff
@@ -167,38 +168,65 @@ def __init__(self, wires, *, shots=None, analytic=None, torch_device=None):
         self._state = self._state.to(self._torch_device)
         self._pre_rotated_state = self._state
 
+    @staticmethod
+    def _get_parameter_torch_device(ops):
+        """An auxiliary function to determine the Torch device specified for
+        the gate parameters of the input operations.
+
+        Returns the first CUDA Torch device found (if any) using a string
+        format. Does not handle tensors put on multiple CUDA Torch devices.
+        Such a case raises an error with Torch.
+
+        If CUDA is not used with any of the parameters, then specifies the CPU
+        if the parameters are on the CPU or None if there were no parametric
+        operations.
+
+        Args:
+            ops (list): list of operations to check
+
+        Returns:
+            str: The string of the Torch device determined.
+        """
+        par_torch_device = None
+        for op in ops:
+            for data in op.data:
+
+                # Using hasattr in case we don't have a Torch tensor as input
+                if hasattr(data, "is_cuda"):
+                    if data.is_cuda:
+                        return ":".join([data.device.type, str(data.device.index)])
```

Comment:
There seems to be a subtlety here, likely because of
```python
In [4]: dev = torch.device("cuda")                                                                                                           

In [5]: str(dev)                                                                                                                             
Out[5]: 'cuda'

In [6]: ":".join([dev.type, str(dev.index)])                                                                                                 
Out[6]: 'cuda:None'
```
This makes 2 tests fail, I'd rather keep it as is. We could think more about the string vs. Torch device storage in the device a bit more in the future.